src/SonarScanner.MSBuild.PreProcessor/WebServer/SonarQubeWebServer.cs

```diff
@@ -41,7 +41,7 @@ public SonarQubeWebServer(IDownloader downloader, Version serverVersion, ILogger
             }
         }
 
-        public override async Task<bool> IsServerLicenseValid()
+        public override async Task<bool> IsServerLicenseValid(string serverUrl)
```

Comment:
It is bizarre to have a parameter here only to be able to log as it could be different from the one used to make the request.
I'd rather have the `IDownloader.GetBaseUri` back and use it to log. That way, we ensure the right `serverUrl` from the `HttpClient`.
WDYT?